cdap-features/src/main/java/io/cdap/cdap/features/Feature.java

```diff
@@ -41,7 +41,8 @@ public enum Feature {
   WRANGLER_PRECONDITION_SQL("6.9.1"),
   WRANGLER_EXECUTION_SQL("6.10.0"),
   WRANGLER_SCHEMA_MANAGEMENT("6.10.0"),
-  NAMESPACED_SERVICE_ACCOUNTS("6.10.0");
+  NAMESPACED_SERVICE_ACCOUNTS("6.10.0"),
+  PIPELINE_PROMOTION_TO_PRODUCTION("6.10.0");
```

Comment:
should not name the feature something vague like this, it should be specific to the actual functionality being controlled. In this case, it is controlling whether bulk source control operations are enabled.

Reply:
Removed the flag from this PR. As mentioned in this [comment](https://github.com/cdapio/cdap/pull/15336#discussion_r1366502823) we do not need feature guards for internal APIs.